src/main/java/io/jenkins/plugins/jfr/JfrModule.java

```diff
@@ -0,0 +1,21 @@
+package io.jenkins.plugins.jfr;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import javax.inject.Singleton;
+
+public class JfrModule extends AbstractModule {
+
+    @Provides
+    @Singleton
+    public JfrService jfrService() {
+        return new JfrServiceImpl();
+    }
+
+    @Provides
+    @Singleton
+    public ObjectMapper objectMapper() {
```

Comment:
Make this a Jakarta named qualifier after the plugin: "Java-flight-recorder". This is important so any other plugin binding an objectmapper does not collide with our definition and leave us in an ambiguous position.